analyzers/tests/SonarAnalyzer.UnitTest/Rules/ClassShouldNotBeEmptyTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using CS = SonarAnalyzer.Rules.CSharp;
+using VB = SonarAnalyzer.Rules.VisualBasic;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class ClassShouldNotBeEmptyTest
+{
+    private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.ClassShouldNotBeEmpty>();
+    private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.ClassShouldNotBeEmpty>();
+
+    [TestMethod]
+    public void ClassShouldNotBeEmpty_CS() =>
+        builderCS
+            .AddPaths("ClassShouldNotBeEmpty.cs")
+            .Verify();
+
+    [TestMethod]
+    public void ClassShouldNotBeEmpty_VB() =>
+        builderVB
+            .AddPaths("ClassShouldNotBeEmpty.vb")
+            .Verify();
+
+#if NET
+
+    private static readonly MetadataReference[] AdditionalReferences = new[]
+    {
+        AspNetCoreMetadataReference.MicrosoftAspNetCoreRazorPages
```

Comment:
what's the behavior with classic ASP .NET MVC (in .NET Framework)? 

because ViewModels are there as well is it the same pattern?

Reply:
(sorry if silly question, you can check afterwards as validation)

Reply:
**ASP.NET MVC** (.NET Framework and .NET Core):
- the Model component: a regular C#/VB.NET class, no need to add it as an exception to the rule.
- the View component is the cshtml/vbhtml file
- the Controller (essentially what the PageModel does in ASP.NET Core Razor Pages): having an empty Controller class makes no sense because then the View isn't rendered at all (unlike in ASP.NET Core Razor Pages). So no need to add it as an exception to the rule.

**ASP.NET Core Razor Pages**: does not support the .NET Framework, and the .NET Core part is already covered in the test cases.